pkg/activator/revision_test.go

```diff
@@ -94,6 +96,130 @@ func TestActiveEndpoint_Reserve_WaitsForReady(t *testing.T) {
 	}
 }
 
+func TestActiveEndpoint_Reserve_WaitsForReady2Step(t *testing.T) {
+	// This generates two updates, so that the inner loop of the `Watch` can be exercised.
+	k8s, kna := fakeClients()
+	kna.ServingV1alpha1().Revisions(testNamespace).Create(
+		newRevisionBuilder(defaultRevisionLabels).
+			withReady(false).
+			build())
+	k8s.CoreV1().Services(testNamespace).Create(newServiceBuilder().build())
+	a := NewRevisionActivator(k8s, kna, TestLogger(t))
+
+	ch := make(chan ActivationResult)
+	go func() {
+		ch <- a.ActiveEndpoint(testNamespace, testRevision)
+	}()
+
+	select {
+	case <-ch:
+		t.Error("Unexpected result before revision is ready.")
+	case <-time.After(1200 * time.Millisecond):
+		// Wait long enough, so that ActiveEndpoint Go routine sets up the Watch.
+		// It does fire any events internally, so we have to "sleep".
+	}
+
+	// Partially update the service, to trigger a watch,
+	// which would not finish the loop.
+	rev, _ := kna.ServingV1alpha1().Revisions(testNamespace).Get(testRevision, metav1.GetOptions{})
+	rev.Status.MarkResourcesAvailable()
+	kna.ServingV1alpha1().Revisions(testNamespace).Update(rev)
+
+	// ... and then finally make revision ready after a timeout.
+	go func() {
+		time.Sleep(250 * time.Millisecond)
+		rev, _ := kna.ServingV1alpha1().Revisions(testNamespace).Get(testRevision, metav1.GetOptions{})
+		rev.Status.MarkActive()
+		rev.Status.MarkContainerHealthy()
+		kna.ServingV1alpha1().Revisions(testNamespace).Update(rev)
+	}()
+
+	select {
+	case ar := <-ch:
+		want := Endpoint{testServiceFQDN, 8080}
```

Comment:
Can you use v1alpha1.DefaultUserPort from revision_types.go?

Reply:
Is it the same one? I just don't know.

Reply:
Added the named constant.